flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamTaskNetworkInput.java

```diff
@@ -73,21 +72,20 @@
 
     private final DeserializationDelegate<StreamElement> deserializationDelegate;
 
-    private final RecordDeserializer<DeserializationDelegate<StreamElement>>[] recordDeserializers;
+    private final Map<InputChannelInfo, RecordDeserializer<DeserializationDelegate<StreamElement>>>
```

Comment:
We are accessing this `Map` just once per buffer, so there should be no performance penalty?

Reply:
Yes, once per buffer. Moreover, map access was already there before:
```
    private void processBuffer(BufferOrEvent bufferOrEvent) throws IOException {
        lastChannel = channelIndexes.get(bufferOrEvent.getChannelInfo());
```
so it shouldn't make any difference.